pouta_blueprints/static/partials/welcome.html

```diff
@@ -1,10 +1,43 @@
-<div class="jumbotron">
-    <h1><configurable-value key="INSTALLATION_NAME"></configurable-value></h1>
-    <p configurable-show key="ENABLE_SHIBBOLETH_LOGIN">
-        <b>You can also login with</b>
-        <a href="/login"><img src="/img/haka.gif" alt="Haka login" /></a>
-    </p>
-    <p>If you don't have access to your password, you
-    can <a href="#/reset_password">reset it here</a>.
-    </p>
+<div class="jumbotron" ng-controller="NavbarController">
+        <div class="col-md-6 col-xs-6">
+            <h1 name="installation-name"><configurable-value key="INSTALLATION_NAME"></configurable-value></h1>
+            <p name="installation-description"><configurable-value
+                                    key="INSTALLATION_DESCRIPTION"></configurable-value></p>
+        </div>
+    <div class="row">
+        <div class="col-md-6 col-xs-6">
+            <p configurable-show key="ENABLE_SHIBBOLETH_LOGIN">
+                <b>Login with with</b>
+                <a href="/login"><img src="/img/haka.gif" alt="Haka login" /></a>
+            </p>
+        </div>
+    </div>
+    <div class="row" ><div class="col-md-6 col-xs-6" ng-show="!getLoginVisibility()">
+        <a class="btn btn-default" ng-click="toggleLoginVisibility()"
+        name="click-show-login">Login with username and password</a>
+        </div>
+    </div>
+    <div class="row" ng-show="getLoginVisibility()">
+        <div class="col-md-6 col-xs-6">
+            <h2>Login</h2>
+            <form ng-submit="login()" >
+              <div ng-show="invalidLogin()" class="form-group">
+                  <span id="invalid-login" class="label label-default"><strong>Invalid login!</strong></span>
+              </div>
+              <div class="form-group">
+                <input type="email" id="email" name="email" ng-model="email" placeholder="Email" class="form-control" required>
+              </div>
+              <div class="form-group">
+                <input type="password" id="password" name="password" ng-model="password" placeholder="Password" class="form-control" required>
+              </div>
+              <div class="form-group">
+                <input type="submit" class="btn btn-success" value="Sign in" />
+              </div>
+            </form>
+            <p>If you don't have access to your password, you
```

Comment:
The reset password text looks quite big and the text overflows in my browser with that column size. I tried getting rid of the p tag to make it smaller, what do you say?